flutter/lib/ui/geometry.dart

```diff
@@ -680,35 +680,183 @@ class Rect {
   String toString() => "Rect.fromLTRB(${left.toStringAsFixed(1)}, ${top.toStringAsFixed(1)}, ${right.toStringAsFixed(1)}, ${bottom.toStringAsFixed(1)})";
 }
 
-// A rounded rectangle with the same radii for all four corners.
+/// A radius for either circular or oval shapes.
+class Radius {
+  /// Constructs a circular radius.
+  const Radius.circular(double radius) : this.oval(radius, radius);
+
+  /// Construct an oval radius.
+  const Radius.oval(this.x, this.y);
+
+  /// A radius with the value zero.
```

Comment:
Maybe add a paragraph here giving an example of how if this is used for an RRect, it leads to a right-angle corner.